crates/json-abi/src/item.rs

```diff
@@ -373,7 +463,63 @@ impl Error {
     }
 }
 
+impl FromStr for Function {
+    type Err = ParserError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::parse(s)
+    }
+}
+
 impl Function {
+    /// Parses a Solidity function signature string:
+    /// `$name($($inputs),*)$(($($outputs),*))?`
+    ///
+    /// Note that [`state_mutability`](Self::state_mutability) is not parsed
+    /// from the input and is always set to [`StateMutability::Payable`].
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// # use alloy_json_abi::{Function, Param, StateMutability};
+    /// assert_eq!(
+    ///     Function::parse("foo(bool bar)"),
+    ///     Ok(Function {
+    ///         name: "foo".to_string(),
+    ///         inputs: vec![Param::parse("bool bar").unwrap()],
+    ///         outputs: vec![],
+    ///         state_mutability: StateMutability::Payable,
+    ///     }),
+    /// )
+    /// ```
+    ///
+    /// [Function]s also support parsing output tuples:
+    ///
+    /// ```
+    /// # use alloy_json_abi::{Function, Param, StateMutability};
+    /// assert_eq!(
+    ///     Function::parse("toString(uint number)(string s)"),
```

Comment:
Do the parsers still work if the string starts with the word `function` for `Function::parse` and `event` for `Event::parse`?

Reply:
No, that should be handled prior. I'll make that clearer, and add a function to `AbiItem` as a way to handle it

Reply:
Done in c947ff7e7a361f32d396d89706a8e4afa8c46d84